File: java-recipes/src/main/java/com/fargopolis/models/Ability.java
package com.fargopolis.models;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import com.fargopolis.enums.AbilitySource;

@Getter
@Setter
@NoArgsConstructor
public class Ability {
    private Integer abilityId;
    private Integer characterId;
    private String name;
    private String description;
    private AbilitySource source;
    private String sourceDescription;
    private String usage;
}

File: recipe-site/app/_models/Ability.ts
import { AbilitySource } from "@/app/_constants/AbilitySource";

export default class Ability {
    abilityId: number;
    characterId: number;
    name: string;
    description: string;
    source: AbilitySource;
    sourceDescription: string;
    usage: string;

    constructor(
        abilityId: number,
        characterId: number,
        name: string,
        description: string,
        source: AbilitySource,
        sourceDescription: string,
        usage: string
    ) {
        this.abilityId = abilityId;
        this.characterId = characterId;
        this.name = name;
        this.description = description;
        this.source = source;
        this.sourceDescription = sourceDescription;
        this.usage = usage;
    }
}
